### PR TITLE
Use sigdescr_np() over sys_siglist (fixes glibc 2.32+ unbundled build)

### PR DESCRIPTION
**Pull request by azat**
glibc 2.32 [1]:

- The functions sigabbrev_np and sigdescr_np have been added
- They should be used instead of sys_siglist or sys_sigabbrev and they
  are both thread and async-signal safe. These functions are GNU
  extensions.

  [1]: https://sourceware.org/pipermail/libc-announce/2020/000029.html

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

*`sys_siglist` introduced in #16366*

### CODE REVIEW

File: base/daemon/BaseDaemon.cpp [hunk 1]
```diff
@@ -85,7 +85,12 @@ const char * msan_strsignal(int sig)
     // Apparently strsignal is not instrumented by MemorySanitizer, so we
     // have to unpoison it to avoid msan reports inside fmt library when we
     // print it.
+#if __GLIBC__ > 2 || (__GLIBC__ == 2  && __GLIBC_MINOR__ >= 32)
```
Comment (alexey-milovidov) [inline base/daemon/BaseDaemon.cpp:88]: AFAIK we are using frozen version of libc headers (see `contrib/libc-headers` and `cmake/toolchain`) on every platform, so there is no chance that `__GLIBC_MINOR__` will change.

Also glibc has no intention to break ABI anytime soon, so `sys_siglist` will continue to work.

Comment (azat) [reply base/daemon/BaseDaemon.cpp:88]: >AFAIK we are using frozen version of libc headers (see contrib/libc-headers and cmake/toolchain) on every platform, so there is no chance that __GLIBC_MINOR__ will change.

For bundled build - yes.

>Also glibc has no intention to break ABI anytime soon, so sys_siglist will continue to work.

ABI - yes, but it does not included into string.h anymore.

>Do I understand correctly that everything is related to the "unbundled" (unfavorable) build?

Yes, this is only for unbundled build

Comment (azat) [reply base/daemon/BaseDaemon.cpp]: I've updated the title to highlight that it is related to unbundled build only.

**Comment by alexey-milovidov** [comment]
> test_concurrent_ttl_merges

> flaky test.

**Comment by alexey-milovidov** [comment]
Fuzzer has found a minor error in the unreleased SimHash functions:
https://github.com/ClickHouse/ClickHouse/issues/18799

May I ask you to take a look, so we will not stop here?

**Comment by azat** [comment]
>Fuzzer has found a minor unrelated error in the unreleased SimHash functions: #18799
>May I ask you to take a look, so we will not stop here?

#18803

**Pull request merged**